portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,42 +13,80 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
-
+import com.google.sps.data.Comment;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.List;
+import java.util.*;
+import java.lang.Integer;
+import com.google.gson.Gson;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that handles comment data. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   
-  private List<String> facts;
+  private static final Gson GSON = new Gson();
 
   @Override
-  public void init() {
-    facts = new ArrayList<>();
-    facts.add(
-        "My favorite color is green. "
-            + "I also like light blue.");
-    facts.add("My parents made up my name based on my Chinese one.");
-    facts.add("My birthday is on Halloween.");
-    facts.add("I have the best younger brother!");
-    facts.add("My favorite food is pasta.");
-    facts.add("I'm a cat person. I like dogs, but I like cats more.");
-    facts.add(
-        "I didn't have any experience with "
-            + "computer science in high school.");
-    facts.add("I play the acoustic guitar.");
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    // Loop over entities.
+    List<Comment> comments = new ArrayList<>();
+    int max = Integer.parseInt(request.getParameter("max"));
+    for (Entity entity : results.asIterable()) {
+      String name = (String) entity.getProperty("name");
+      String text = (String) entity.getProperty("text");
+      long timestamp = (long) entity.getProperty("timestamp");
+      comments.add(new Comment(name, text, timestamp));
+      if (comments.size() == max) {
+        break;
+      }
+    }
+
+    response.setContentType("application/json;");
+    response.getWriter().println(GSON.toJson(comments));
   }
 
-  @Override
```

Comment:
let's keep the "Override" annotation since doPost is also overriding a method:
https://stackoverflow.com/questions/94361/when-do-you-use-javas-override-annotation-and-why